casbah/src/main/scala/akka/contrib/persistence/mongodb/CasbahPersistenceJournaller.scala

```diff
@@ -23,47 +29,73 @@ class CasbahPersistenceJournaller(driver: CasbahMongoDriver) extends MongoPersis
   private[this] def clearEmptyDocumentsQuery(pid: String): DBObject =
     (PROCESSOR_ID $eq pid) ++ (EVENTS $size 0)
 
-  private[this] def journal(implicit ec: ExecutionContext) = driver.journal
+  private[this] def journal(implicit ec: ExecutionContext): MongoCollection = driver.journal
 
   private[this] def realtime(implicit ec: ExecutionContext) = driver.realtime
 
   private[this] def metadata(implicit ec: ExecutionContext) = driver.metadata
 
-  private[mongodb] def journalRange(pid: String, from: Long, to: Long)(implicit ec: ExecutionContext): Iterator[Event] =
+  private[mongodb] def journalRange(pid: String, from: Long, to: Long)(implicit ec: ExecutionContext): Iterator[Event] = {
+    val journal = driver.getJournal(pid)
     journal.find(journalRangeQuery(pid, from, to))
-            .sort(MongoDBObject(TO -> 1))
-           .flatMap(_.getAs[MongoDBList](EVENTS))
-           .flatMap(lst => lst.collect { case x:DBObject => x })
-           .filter(dbo => dbo.getAs[Long](SEQUENCE_NUMBER).exists(sn => sn >= from && sn <= to))
-           .map(driver.deserializeJournal)
-
-  import collection.immutable.{Seq => ISeq}
-  private[mongodb] override def batchAppend(writes: ISeq[AtomicWrite])(implicit ec: ExecutionContext):Future[ISeq[Try[Unit]]] = Future {
+      .sort(MongoDBObject(TO -> 1))
+      .flatMap(_.getAs[MongoDBList](EVENTS))
+      .flatMap(lst => lst.collect { case x: DBObject => x })
+      .filter(dbo => dbo.getAs[Long](SEQUENCE_NUMBER).exists(sn => sn >= from && sn <= to))
+      .map(driver.deserializeJournal)
+  }
+
+  import collection.immutable.{ Seq => ISeq }
+  private[this] def doBatchJournalAppend(writes: ISeq[AtomicWrite], journal: MongoCollection)(implicit ec: ExecutionContext): ISeq[Try[Unit]] = {
     val batch = writes.map(write => Try(driver.serializeJournal(Atom[DBObject](write, driver.useLegacySerialization))))
+    
     if (batch.forall(_.isSuccess)) {
       val bulk = journal.initializeOrderedBulkOperation
       batch.collect { case scala.util.Success(ser) => ser } foreach bulk.insert
       bulk.execute(writeConcern)
-
-      if(driver.realtimeEnablePersistence) {
-        val bulk2 = realtime.initializeOrderedBulkOperation
-        batch.collect { case scala.util.Success(ser) => ser } foreach bulk2.insert
-        bulk2.execute(writeConcern)
-      }
-
       batch.map(t => t.map(_ => ()))
-    } else { // degraded performance, cant batch
-      batch.map(_.map{serialized =>
+    } else { // degraded performance, can't batch
+      batch.map(_.map { serialized =>
         journal.insert(serialized)(identity, writeConcern)
-        if(driver.realtimeEnablePersistence) realtime.insert(serialized)(identity, writeConcern)
+        //if (driver.realtimeEnablePersistence) realtime.insert(serialized)(identity, writeConcern)
```

Comment:
Can you delete these commented out lines?  I've seen a few so far, so if you could make a pass all the way through that would be nice.